lib/mongoidable/ability_updater.rb

```diff
@@ -39,12 +39,12 @@ def database_ability
 
     def should_update?
       # If the ability change includes model attributes new up a model with those attributes to check
-      arguments = if subject_as_class.is_a?(Symbol) || extra.blank?
-                    [action, subject_as_class]
-                  else
-                    [action, subject_as_class, extra.first]
-                  end
-      parent_document.current_ability.can?(*arguments) != base_behavior
+      can_args = if subject_as_class.is_a?(Symbol) || extra.blank?
+                   [action, subject_as_class]
+                 else
+                   [action, subject_as_class.new(extra.first)]
```

Comment:
Found a way to break this:

(Bad pseudo code, but the idea is mostly there...)
```
class MyModel do
  field :breaking_field, type: Array
end

it "does not re-add permissions on breaking_field" do
  policy = FactoryBot.create :policy
  policy.instance_abilities.create! base_behavior: true, action: :an_action, subject: MyModel, extra: [{ breaking_field: "A Value" }]

  expect do
    policy.instance_abilities.add_ability base_behavior: true, action: :an_action, subject: MyModel, extra: [{ breaking_field: "A Value" }]
  end.not_to(change { policy.reload.instance_abilities.count })

  expect(policy.current_ability.can?(:an_action, MyModel)).to be_true
end
```

This call `subject_as_class.new(extra.first)` cannot massage the string into an array, so the assignment fails, and you get an object with a value of `breaking_field: []` which doesn't match the condition giving us a false-negative, causing the code to think that the condition does not match, and you `cannot?`, not `can?`.